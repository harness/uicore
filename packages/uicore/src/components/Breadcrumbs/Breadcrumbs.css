/*
 * Copyright (c) Harness Inc.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

.breadcrumbs {
  line-height: 18px;
  .breadcrumb {
    display: flex;
    align-items: center;
    justify-content: flex-start;
  }
}
