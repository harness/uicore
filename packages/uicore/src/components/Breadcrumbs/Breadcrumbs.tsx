/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Link } from 'react-router-dom'
import cx from 'classnames'
import { Icon, IconProps } from '@harness/icons'
import { Layout } from '../../layouts/Layout'
import { Text } from '../Text/Text'
import css from './Breadcrumbs.css'

export interface Breadcrumb {
  url: string
  label: string
  iconProps?: IconProps
  onClick?: () => void
}
export interface BreadcrumbsProps {
  links: Breadcrumb[]
  className?: string
}

export const Breadcrumbs: React.FC<BreadcrumbsProps> = ({ links = [], className = '' }) => {
  return (
    <Layout.Horizontal className={cx(css.breadcrumbs, className)}>
      {links.map((link: Breadcrumb) => {
        return (
          <Layout.Horizontal flex={{ align: 'center-center', justifyContent: 'flex-start' }} key={link.label}>
            <Link
              to={link.url}
              onClick={event => {
                if (link.onClick) {
                  event.preventDefault()
                  link.onClick()
                }
              }}>
              {link.iconProps && link.iconProps.name && (
                <Icon size={16} padding={{ right: 'xsmall' }} {...link.iconProps} />
              )}
              <Text intent="primary" font={{ size: 'small' }}>
                {link.label}
              </Text>
            </Link>
            <Icon size={8} name="main-chevron-right" color="grey500" padding={{ right: 'xsmall', left: 'xsmall' }} />
          </Layout.Horizontal>
        )
      })}
    </Layout.Horizontal>
  )
}
