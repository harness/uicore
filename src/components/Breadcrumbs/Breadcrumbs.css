.breadcrumbs {
  line-height: 18px;
  .breadcrumb {
    display: flex;
    align-items: center;
    justify-content: flex-start;
  }
}
