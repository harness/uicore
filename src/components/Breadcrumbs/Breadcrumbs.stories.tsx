import React from 'react'
import type { Meta, Story } from '@storybook/react'
import { BrowserRouter } from 'react-router-dom'

import { BreadcrumbsProps, Breadcrumbs } from '../..'
import { Title, Subtitle, ArgsTable, Stories, PRIMARY_STORY, Primary } from '@storybook/addon-docs/blocks'

import { Icon } from '../../icons/Icon'

export default {
  title: 'Components / Breadcrumbs',
  subcomponents: { IconProps: Icon },
  component: Breadcrumbs,

  parameters: {
    layout: 'centered',
    docs: {
      source: {
        type: 'code'
      },

      page: function PageDescription() {
        return (
          <>
            <Title>Breadcrumbs</Title>

            <Subtitle>
              <pre>
                <code>{`
import { Breadcrumbs }  from '@wings-software/uicore'
export interface Breadcrumb {
  url: string
  label: string
  iconProps?: IconProps
}`}</code>
              </pre>
            </Subtitle>
            <Primary />
            <ArgsTable story={PRIMARY_STORY} />

            <Stories />
          </>
        )
      }
    }
  },
  decorators: [Story => <Story />]
} as Meta
export const Basic: Story<BreadcrumbsProps> = args => {
  const { links, className } = args
  return (
    <BrowserRouter>
      <Breadcrumbs links={links} className={className} />
    </BrowserRouter>
  )
}
Basic.args = {
  links: [
    {
      url: '#',
      label: 'Google',
      iconProps: { name: 'add', color: 'green500' }
    },
    {
      url: '#',
      label: 'Gmail',
      iconProps: { name: 'pipeline-approval', color: 'green500' }
    },
    {
      url: '#',
      label: 'Gmail2'
    }
  ]
}
