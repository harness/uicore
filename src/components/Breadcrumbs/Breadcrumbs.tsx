import React from 'react'
import { Link } from 'react-router-dom'
import cx from 'classnames'
import { Icon, IconProps } from '../../icons/Icon'
import { Layout } from '../../layouts/Layout'
import { Text } from '../Text/Text'
import css from './Breadcrumbs.css'

export interface Breadcrumb {
  url: string
  label: string
  iconProps?: IconProps
}
export interface BreadcrumbsProps {
  links: Breadcrumb[]
  className?: string
}

export const Breadcrumbs: React.FC<BreadcrumbsProps> = ({ links = [], className = '' }) => {
  return (
    <Layout.Horizontal
      flex={{ align: 'center-center', justifyContent: 'flex-start' }}
      className={cx(css.breadcrumbs, className)}>
      {links.map((link: Breadcrumb, index: number) => (
        <Layout.Horizontal flex={{ align: 'center-center', justifyContent: 'flex-start' }} key={index}>
          <Link className={css.breadcrumb} to={link.url}>
            {link.iconProps && link.iconProps.name && (
              <Icon size={16} padding={{ right: 'xsmall' }} {...link.iconProps} />
            )}
            <Text intent="primary" font={{ size: 'small' }}>
              {link.label}
            </Text>
          </Link>
          <Icon size={8} name="main-chevron-right" color="grey500" padding={{ right: 'xsmall', left: 'xsmall' }} />
        </Layout.Horizontal>
      ))}
    </Layout.Horizontal>
  )
}
