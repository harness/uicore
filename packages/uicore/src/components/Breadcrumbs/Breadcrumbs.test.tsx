/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { Breadcrumbs } from '../..'
import { Breadcrumb } from './Breadcrumbs'

import { BrowserRouter } from 'react-router-dom'

describe('Render basic component', () => {
  test('should check snapshot with name', () => {
    const links: Breadcrumb[] = [
      {
        url: '#',
        label: 'Google',
        iconProps: { name: 'add' }
      },
      {
        url: '#',
        label: 'Gmail',
        iconProps: { name: 'pipeline-approval' }
      },
      {
        url: '#',
        label: 'Gmail2'
      }
    ]
    const { container } = render(
      <BrowserRouter>
        <Breadcrumbs links={links} />
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
