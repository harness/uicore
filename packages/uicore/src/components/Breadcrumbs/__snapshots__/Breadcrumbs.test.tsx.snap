// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render basic component should check snapshot with name 1`] = `
<div>
  <div
    class="horizontal layout-spacing-undefined StyledProps--main breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
  >
    <div
      class="horizontal layout-spacing-undefined StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
    >
      <a
        class="breadcrumb"
        href="/"
      >
        <span
          class="bp3-icon bp3-icon-add StyledProps--main StyledProps--padding-right-xsmall"
          icon="add"
        >
          <svg
            data-icon="add"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              add
            </desc>
            <path
              d="M10.99 6.99h-2v-2c0-.55-.45-1-1-1s-1 .45-1 1v2h-2c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1v-2h2c.55 0 1-.45 1-1s-.45-1-1-1zm-3-7c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.68 6-6 6z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <p
          class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
        >
          Google
        </p>
      </a>
      <span
        class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
        data-icon="main-chevron-right"
      >
        <svg
          height="8"
          viewBox="0 0 8 14"
          width="8"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
            fill="#4d5969"
          />
        </svg>
      </span>
    </div>
    <div
      class="horizontal layout-spacing-undefined StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
    >
      <a
        class="breadcrumb"
        href="/"
      >
        <span
          class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
          data-icon="pipeline-approval"
        >
          <svg
            height="16"
            viewBox="0 0 19 17"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.81.669l.006.004.015.013a.5.5 0 01.062.7L6.479 16.375a.5.5 0 01-.709.062.52.52 0 01-.136-.176.523.523 0 01-.126-.095L.132 10.578a.5.5 0 01.005-.698l.017-.018a.5.5 0 01.707-.005l.007.007 5.233 5.367L18.107.735a.5.5 0 01.704-.066z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <p
          class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
        >
          Gmail
        </p>
      </a>
      <span
        class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
        data-icon="main-chevron-right"
      >
        <svg
          height="8"
          viewBox="0 0 8 14"
          width="8"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
            fill="#4d5969"
          />
        </svg>
      </span>
    </div>
    <div
      class="horizontal layout-spacing-undefined StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
    >
      <a
        class="breadcrumb"
        href="/"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
        >
          Gmail2
        </p>
      </a>
      <span
        class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
        data-icon="main-chevron-right"
      >
        <svg
          height="8"
          viewBox="0 0 8 14"
          width="8"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
            fill="#4d5969"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
